src/components/Search/SearchAutocompleteInput.tsx

```diff
@@ -174,11 +174,19 @@ function SearchAutocompleteInput(
     const clearFilters = useCallback(() => {
         clearAdvancedFilters();
         onSearchQueryChange('');
-        Navigation.navigate(
-            ROUTES.SEARCH_ROOT.getRoute({
-                query: buildCannedSearchQuery(),
-            }),
-        );
+
+        // Check if we are on the search page before clearing query. If we are using the popup search menu,
+        // then the clear button is ONLY available when the search is *not* saved, so we don't have to navigate
+        const currentRoute = Navigation.getActiveRouteWithoutParams();
+        const isSearchPage = currentRoute === `/${ROUTES.SEARCH_ROOT.route}`;
+
```

Comment:
Let's reuse the `isOnSearch` value provided by `useSearchContext`, e.g. [here](https://github.com/Expensify/App/blob/8fb70c33296f023d567791c2922787364268dab2/src/components/ReportActionItem/ReportPreview.tsx#L179)